core/server/worker/src/main/java/alluxio/worker/page/PagedBlockWriter.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.worker.page;
+
+import alluxio.client.file.CacheContext;
+import alluxio.client.file.cache.CacheManager;
+import alluxio.client.file.cache.PageId;
+import alluxio.network.protocol.databuffer.DataBuffer;
+import alluxio.worker.block.io.BlockWriter;
+
+import io.netty.buffer.ByteBuf;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.WritableByteChannel;
+
+/**
+ * A paged implementation of BlockWriter interface.
+ */
+public class PagedBlockWriter extends BlockWriter {
+  private static final Logger LOG = LoggerFactory.getLogger(PagedBlockWriter.class);
+
+  private final CacheManager mCacheManager;
+  private final String mBlockId;
+  private final long mPageSize;
+  private long mPosition;
+
+  PagedBlockWriter(CacheManager cacheManager, long blockId, long pageSize) {
+    mCacheManager = cacheManager;
+    mBlockId = String.valueOf(blockId);
+    mPageSize = pageSize;
+  }
+
+  @Override
+  public long append(ByteBuffer inputBuf) throws IOException {
+    long bytesWritten = 0;
+    while (inputBuf.hasRemaining()) {
+      PageId pageId = getPageId(bytesWritten);
+      int currentPageOffset = getCurrentPageOffset();
+      int bytesLeftInPage = getBytesLeftInPage(currentPageOffset, inputBuf.remaining());
+      byte[] page = new byte[bytesLeftInPage];
+      inputBuf.get(page);
```

Comment:
Is there a way to restructure the two function to call a common `appendInternal` to reduce duplicated logic?